### PR TITLE
Add semantic properties to DialogWindow, Popup and Dialog

### PR DESCRIPTION
**Pull request by m-sasha**
## Proposed Changes

  - Add `dialog()` semantic property to `DialogWindow`.
  - Add `dialog()` semantic property to `Dialog`
  - Add `popup()` semantic property to `Popup`.
  

## Testing

Test: Added (and enabled) unit tests for this.

### CODE REVIEW

File: compose/ui/ui-test-junit4/src/desktopTest/kotlin/androidx/compose/ui/test/FiltersTest.kt [hunk 1]
```diff
@@ -72,13 +72,29 @@ class FiltersTest {
...
 
+        rule.onNodeWithTag("tag").assert(hasAnyAncestor(isDialog()))
+    }
+
+    @Test
+    fun testIsPopup() {
+        rule.setContent {
+            Popup(
+
+            ) {
```
Comment (MatkovIvan) [inline compose/ui/ui-test-junit4/src/desktopTest/kotlin/androidx/compose/ui/test/FiltersTest.kt]:
```suggestion
            Popup {
```

File: compose/ui/ui-test-junit4/src/desktopTest/kotlin/androidx/compose/ui/test/FiltersTest.kt [hunk 2]
```diff
@@ -72,13 +72,29 @@ class FiltersTest {
...
+                Text(
+                    text = "Text",
+                    modifier = Modifier.testTag("tag1")
+                )
+            }
+        }
+
         rule.onNodeWithTag("tag").assert(hasAnyAncestor(isPopup()))
```
Comment (MatkovIvan) [inline compose/ui/ui-test-junit4/src/desktopTest/kotlin/androidx/compose/ui/test/FiltersTest.kt:89]: It has `testTag("tag1")`. Does it work? why?

File: compose/ui/ui-test-junit4/src/desktopTest/kotlin/androidx/compose/ui/test/TextActionsTest.kt [hunk 3]
```diff
@@ -51,7 +51,7 @@ class TextActionsTest {
             TestTextField("hello")
         }
 
-        with(rule.onNodeWithTag("tag")){
+        with(rule.onNodeWithTag("tag")) {
```
Comment (MatkovIvan) [inline compose/ui/ui-test-junit4/src/desktopTest/kotlin/androidx/compose/ui/test/TextActionsTest.kt:54]: It looks like only whitespace changed in this file

File: compose/ui/ui-test-junit4/src/desktopTest/kotlin/androidx/compose/ui/test/FiltersTest.kt (hunk 2 above)
Comment (m-sasha) [reply compose/ui/ui-test-junit4/src/desktopTest/kotlin/androidx/compose/ui/test/FiltersTest.kt:89]: Started changing it to test both `Popup` overloads, and then decided not to.

File: compose/ui/ui-test-junit4/src/desktopTest/kotlin/androidx/compose/ui/test/TextActionsTest.kt (hunk 3 above)
Comment (m-sasha) [reply compose/ui/ui-test-junit4/src/desktopTest/kotlin/androidx/compose/ui/test/TextActionsTest.kt:54]: Yes, I fixed it here and in a bunch of other test files.

**Review by MatkovIvan — APPROVED** [review]

**Review by igordmn — APPROVED** [review]

**Review by MatkovIvan — COMMENTED** [review]
Should we reuse 
https://github.com/JetBrains/compose-multiplatform-core/blob/55f3f01feda73cc4c0dcf7295942f59029579b92/compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/platform/SkiaBasedOwner.skiko.kt#L108
?

UPD: Will be fixed in separate PR

**Pull request merged**